src/DotNetty.Transport/Channels/Sockets/DatagramChannel.cs

```diff
@@ -0,0 +1,460 @@
+
+        readonly DefaultDatagramChannelConfig config;
+
+        public DatagramChannel()
+            : this(new Socket(SocketType.Dgram, ProtocolType.Udp))
+        {
+        }
+
+        public DatagramChannel(AddressFamily addressFamily)
+            : this(new Socket(addressFamily, SocketType.Dgram, ProtocolType.Udp))
+        {
+        }
+
+        public DatagramChannel(Socket socket)
+            : base(null, socket)
+        {
+            this.config = new DefaultDatagramChannelConfig(this, socket);
+
+            // Read is not allowed until socket is bound for UDP
+            this.config.AutoRead = false;
+        }
+
+        public override IChannelConfiguration Configuration => this.config;
+
+        public override ChannelMetadata Metadata => ChannelMetadata;
+
+        protected override EndPoint LocalAddressInternal => this.Socket.LocalEndPoint;
+
+        protected override EndPoint RemoteAddressInternal => this.Socket.RemoteEndPoint;
+
+        protected override void DoBind(EndPoint localAddress)
+        {
+            this.Socket.Bind(localAddress);
+            this.CacheLocalAddress();
+
+            this.SetState(StateFlags.Active);
+            this.config.AutoRead = true;
+        }
+
+        public override bool Active => 
+            this.Open
+            && (this.config.ActiveOnOpen && this.Registered || this.Socket.IsBound);
+
+        protected override bool DoConnect(EndPoint remoteAddress, EndPoint localAddress)
+        {
+            if (localAddress != null)
+            {
+                this.DoBind(localAddress);
+            }
+
+            bool success = false;
+            try
+            {
+                this.Socket.Connect(remoteAddress);
+                success = true;
+                return true;
+            }
+            finally
+            {
+                if (!success)
+                {
+                    this.DoClose();
+                }
+            }
+        }
+
+        protected override void DoFinishConnect(SocketChannelAsyncOperation operation)
+        {
+            throw new NotSupportedException();
+        }
+
+        protected override void DoDisconnect() => this.DoClose();
+
+        protected override void DoClose()
+        {
+            if (this.TryResetState(StateFlags.Open | StateFlags.Active))
+            {
+                this.Socket.Shutdown(SocketShutdown.Both);
+                this.Socket.Dispose();
+            }
+        }
+
+        protected override void ScheduleSocketRead()
+        {
+            SocketChannelAsyncOperation operation = this.ReadOperation;
+            operation.RemoteEndPoint = new IPEndPoint(
+                this.LocalAddress.AddressFamily == AddressFamily.InterNetwork ? IPAddress.Any : IPAddress.IPv6Any,
+                IPEndPoint.MinPort);
+
+            IRecvByteBufAllocatorHandle handle = this.Unsafe.RecvBufAllocHandle;
+            IByteBuffer buffer = handle.Allocate(this.config.Allocator);
```

Comment:
I believe the default maximum for UDP is 65536. It will be discarded since UDP does not offer flow control like TCP. What is the best way to handle this?